Objects/typeobject.c

```diff
@@ -309,7 +296,13 @@ PyType_ClearCache(void)
 void
 _PyType_Fini(PyInterpreterState *interp)
 {
-    _PyType_ClearCache(interp);
+    struct type_cache *cache = &interp->type_cache;
+    for (Py_ssize_t i = 0; i < (1 << MCACHE_SIZE_EXP); i++) {
+        struct type_cache_entry *entry = &cache->hashtable[i];
+        entry->version = 0;
+        Py_CLEAR(entry->name);
+        entry->value = NULL;
+    }
```

Comment:
Why not reusing type_cache_clear() with a parameter to decide how to handle entry->name, to factorize the code?

Reply:
Because `type_cache_clear()` does something different. It resets the cache to its starting state.
This code clears it out during finalization.

Any function with a boolean argument that changes it behavior is really two functions.

Reply:
I've refactored `type_cache_clear()` to take a `PyObject *`.